### PR TITLE
Pytest-ify CleanCat tests

### PR DESCRIPTION
**Pull request by wojcikstefan**
This PR:
* Increases coverage from 56% to 90% (tested locally on Py2.7 via `pytest-cov`).
* Converts old unittest-style tests based on inheritance into pytest-style tests based on parameters and composable fixtures.
* Turns many schema-level tests into field-level tests.
* Asserts contents of error messages, not just that *any* error occurred. This surfaced some issues/inconsistencies for which I left TODOs.
* Cleans up `setup.cfg`.
* Removes no-longer-used `cleancat.utils`. That module used to contain test-specific tooling that we get for free with pytest (thanks to pretty diffing of regular `assert`s).

Commits can be reviewed separately.

### CODE REVIEW

File: tests/test_base.py [hunk 1]
```diff
@@ -0,0 +1,941 @@
...
+        ('short', False),
+    ])
+    def test_it_enforces_min_length(self, value, valid):
+        field = String(min_length=10)
+        if valid:
+            assert field.clean(value) == value
+        else:
+            with pytest.raises(ValidationError) as e:
+                field.clean(value)
+            assert e.value.args[0] == (
```
Comment (tsx) [inline tests/test_base.py]: `with pytest.raises(..., message='The value must be ...')`

Comment (wojcikstefan) [reply tests/test_base.py]: [Docs for `pytest.raises`](https://docs.pytest.org/en/latest/reference.html#pytest-raises) say:

> if specified, provides a custom failure message if the exception is not raised

It's not to assert what the exception's message is.

Comment (wojcikstefan) [reply tests/test_base.py]: I guess `match` is what we could use. That said, I don't think it provides much value over this assert.

Comment (tsx) [reply tests/test_base.py]: Ah yep, I mixed up keywords. `match` it is.

Comment (tsx) [reply tests/test_base.py]: Ah yep, I mixed up keywords. `match` it is.

Comment (tsx) [reply tests/test_base.py]: Ah yep, I mixed up keywords. `match` it is.

Comment (tsx) [reply tests/test_base.py]: Ah yep, I mixed up keywords. `match` it is.

Comment (wojcikstefan) [reply tests/test_base.py]: Done in 5027de2

**Comment by wojcikstefan** [comment]
Yep, all old tests are still covered.

**Pull request merged**